US-14: Indoor Points of Interest

cristian-aldea opened the pull request:
This PR aims to implements US-14 (#19).

Two new components, CustomMarker and FloorPicker, were created to allow for rendering of markers with custom text and to give the user control over the displayed building

Building Highlights has been refactored to MapOverlays. All overlays (markers, polygons) that are drawn over the map are present in this component.

Dependencies were updated.

Global/common styling was refactored into global constants where possible

cristian-aldea committed:
Add Hall indoor map floors and POIs #19

The procedure for adding indoor floors was started.

Also, indoor POIs were added for each floor which will have indoor maps

Co-authored-by: Cindy <lo_cindy@outlook.com>
Co-authored-by: Omar <osahtout@hotmail.com>
Co-authored-by: Yunshi <dabris16.x@gmail.com>

ys-lin committed:
add Loyola CC1 floor overlay and POI #19

ys-lin committed:
update snap test #19

cristian-aldea committed:
Change zoom level handling and refactor #19

Added markers for each campus

Added zoom levels which will allow us to filter overlays and markers
based on zoom level.

Moved floor data to its own file

cindyslittleplanet committed:
Specified region according to ZoomLevel #19

cristian-aldea committed:
Custom marker and filter by zoom level #19

cristian-aldea committed:
Update dependencies #19

osahtout committed:
Implement JMSB floor and POIs #19

cristian-aldea committed:
Refactor Overlays and overlay filtering #19

All overlays (markers, polygons) will now be rendered from one
component, where proper filtering based on context is done.

cristian-aldea committed:
Update documentation and removed useless code #19

cristian-aldea committed:
Implement and fix tests #19

osahtout committed:
fix ios render bug and update snap #19

bit172 commented on app.json:
```diff
@@ -3,7 +3,6 @@
     "name": "WayFinder",
     "slug": "expo-init-project",
     "privacy": "public",
-    "sdkVersion": "36.0.0",
```
Should this be removed?

cristian-aldea replied:
Expo removed this line when I did `expo update`. I'm not sure what the point of it is, but I guess it makes expo use the latest sdk

bit172 replied:
Ah, I see

JimLau49 reviewed (changes requested):
### DoD
- [x] Documentation present for every component and function
- [x] Unit and snapshot tests are implemented for every component and function
- [x] All tests pass
- [x] Implementation matches the corresponding mockup from Figma

### ACCEPTANCE CRITERIA
- [x] I should be able to see different Points of Interests (POIs) and floors inside the building.
- [x] I should be able to control the floor I'm seeing on the map
- [ ] POIs should only show up on the indoor map if the floor they reside in is selected.
- [x] POIs that are shown should be classes, bathrooms, elevators, stairs, etc.

### USER FLOW ACCEPTANCE TEST
- [x] User is able to freely switch between floors using the floor picker
- [x] User sees different POIs on each floor
- [ ] POIs seen are: classes, bathrooms ,elevators, stairs, etc

**BLOCKING**:
- [iOS] it seems that when using the floor picker the POI's for 9th floor still have 8th floor rooms as indicated by H8__. 
![image](https://user-images.githubusercontent.com/35787279/76982859-de696b80-6912-11ea-8d24-4220a1369436.png)

- [iOS] Buildings appear as blue after zooming in and out once on 
![image](https://user-images.githubusercontent.com/35787279/76982835-d7425d80-6912-11ea-90fc-2d61f9648daf.png)

RGPosadas commented on src/components/floor-picker/floor-picker.component.tsx:
```diff
@@ -0,0 +1,82 @@
+  onFloorPickerButtonPress: (index: number) => void;
+}
+
+const FloorPicker = ({
```
Needs documentation

leojrsilao reviewed (commented):
### DoD
- [x] Documentation present for every component and function
- [x] Unit and snapshot tests are implemented for every component and function
- [x] All tests pass
- [x] Implementation matches the corresponding mockup from Figma

### ACCEPTANCE CRITERIA
- [x] I should be able to see different Points of Interests (POIs) and floors inside the building.
- [x] I should be able to control the floor I'm seeing on the map
- [ ] POIs should only show up on the indoor map if the floor they reside in is selected.
- [x] POIs that are shown should be classes, bathrooms, elevators, stairs, etc.
- [x] Must match Mockup
- [x] Respect the Definition of Done

### USER FLOW ACCEPTANCE TEST
- [x] Pre-requisite: user must be zoomed in enough on the map to enable indoor view
- [x] User is able to freely switch between floors using the floor picker
- [x] User sees different POIs on each floor
- [ ] POIs seen are: classes, bathrooms ,elevators, stairs, etc

**BLOCKING**:
- [iOS] Switching to the 9th floor will still have the POI values of the 8th floor.
![image](https://user-images.githubusercontent.com/35787279/76982859-de696b80-6912-11ea-8d24-4220a1369436.png)

- [iOS] Buildings appear as blue after zooming in and out.
![image](https://user-images.githubusercontent.com/35787279/76982835-d7425d80-6912-11ea-90fc-2d61f9648daf.png)

RGPosadas commented on src/constants/poi.data.ts:
```diff
@@ -1459,5 +1494,408 @@ export const POIInfo: POI[] = [
+  },
+  {
+    id: null,
+    displayName: "Stars to second floor",
```
Typo for stairs

cristian-aldea replied:
@osahtout @cindyslittleplanet @ys-lin Can you fix the typos in the displaynames for the POIs? and also make them shorter? If not I'll tackle it later today

cindyslittleplanet replied:
Yes, I'm gonna go over the POIs

RGPosadas reviewed (changes requested):
### DoD
- [ ] Documentation present for every component and function (see [comment](https://github.com/RGPosadas/WayFinder/pull/57#discussion_r394483013))
- [x] Unit and snapshot tests are implemented for every component and function
- [x] All tests pass (see Question #1)
- [x] Implementation matches the corresponding mockup from Figma

### ACCEPTANCE CRITERIA
- [x] I should be able to see different Points of Interests (POIs) and floors inside the building.
- [x] I should be able to control the floor I'm seeing on the map
- [x] POIs should only show up on the indoor map if the floor they reside in is selected.
- [x] POIs that are shown should be classes, bathrooms, elevators, stairs, etc.

### USER FLOW ACCEPTANCE TEST
- [x] Pre-requisite: user must be zoomed in enough on the map to enable indoor view
- [x] User is able to freely switch between floors using the floor picker
- [x] User sees different POIs on each floor
- [x] POIs seen are: classes, bathrooms ,elevators, stairs, etc

* QUESTION #1: Have you investigated this? I know the tests pass however, but the fact that it's an error log must be dealt with since it can also be seen on Travis.
![image](https://user-images.githubusercontent.com/33991121/76983040-24263400-6913-11ea-9e55-309c1750cc19.png)
* Notes: Tested on Google Pixel 3

r-saba commented on __tests__/custom-marker.test.js:
```diff
@@ -0,0 +1,61 @@
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("should run function on press", () => {
```
Not descriptive enough

TheGreatMarkus commented:
> * QUESTION #1: Have you investigated this? I know the tests pass however, but the fact that it's an error log must be dealt with since it can also be seen on Travis.
>   ![image](https://user-images.githubusercontent.com/33991121/76983040-24263400-6913-11ea-9e55-309c1750cc19.png)
> * Notes: Tested on Google Pixel 3

I've been investigating the issue without luck. It's caused by the setTimeOut in the map overlay useEffect hook. I will continue researching to see if it is possible to fix this.
https://github.com/testing-library/react-testing-library/issues/281
![image](https://user-images.githubusercontent.com/23544999/76987948-60a95e00-691a-11ea-8b84-1ddc89207243.png)

cristian-aldea committed:
Address some pr feedback #19

r-saba commented on src/components/custom-marker/custom-marker.component.tsx:
```diff
@@ -0,0 +1,103 @@
+
+  return (
+    <Marker
+      testID={"marker"}
```
Isn't this a code smell? Our source code has code specifically for testing

cristian-aldea replied:
Yes it is technically a test smell to have these ids that will only be used for testing. However, I discussed with @RGPosadas  it is the most reliable and easy way that enzyme provides to find elements for testing.

RGPosadas replied:
@r-saba It's specifically for Detox, not necessarily Enzyme (tho it would benefit Enzyme with finding props)
It's also a react native thing specifically for e2e testing
Sources: https://reactnative.dev/docs/view.html#testid

TheGreatMarkus commented:
@JimLau49 @leojrsilao Thanks for mentioning your issues on ios, I wasn't having these issues during development but they must be addressed. I will investigate possible solutions.

r-saba commented on src/screens/map-screen/map-screen.component.tsx:
```diff
@@ -100,15 +133,75 @@ const MapScreen = () => {
+  /**
+   * Handles react-native-maps events for indoor floors.
+   *
+   * @param event event object for indoor floor entry
```
typo 

> event event

r-saba commented:
### DoD
- [X] Documentation present for every component and function
- [X] Unit and snapshot tests are implemented for every component and function
- [X] All tests pass
- [X] Implementation matches the corresponding mockup from Figma

### ACCEPTANCE CRITERIA
- [X] I should be able to see different Points of Interests (POIs) and floors inside the building.
- [X] I should be able to control the floor I'm seeing on the map
- [X] POIs should only show up on the indoor map if the floor they reside in is selected.
- [X] POIs that are shown should be classes, bathrooms, elevators, stairs, etc.

### USER FLOW ACCEPTANCE TEST
- [X] Pre-requisite: user must be zoomed in enough on the map to enable indoor view
- [X] User is able to freely switch between floors using the floor picker
- [X] User sees different POIs on each floor
- [X] POIs seen are: classes, bathrooms ,elevators, stairs, etc

bit172 commented on src/constants/style.ts:
```diff
@@ -1,7 +1,18 @@
+export const BUILDING_MARKER_COLOR = "#252525";
+
+// Dimensions
+export const screenHeight = Dimensions.get("window").height;
```
Nice to see these in a separate file.

bit172 reviewed (dismissed):
### DoD
- [x] Documentation present for every component and function
- [x] Unit and snapshot tests are implemented for every component and function
- [x] All tests pass
- [x] Implementation matches the corresponding mockup from Figma

### ACCEPTANCE CRITERIA
- [x] I should be able to see different Points of Interests (POIs) and floors inside the building.
- [x] I should be able to control the floor I'm seeing on the map
- [x] POIs should only show up on the indoor map if the floor they reside in is selected.
- [x] POIs that are shown should be classes, bathrooms, elevators, stairs, etc.

### USER FLOW ACCEPTANCE TEST
- [x] Pre-requisite: user must be zoomed in enough on the map to enable indoor view
- [x] User is able to freely switch between floors using the floor picker
- [x] User sees different POIs on each floor
- [x] POIs seen are: classes, bathrooms ,elevators, stairs, etc

cindyslittleplanet committed:
Shorten POIs and fixed POIs typos

bit172 commented on src/constants/poi.data.ts:
```diff
@@ -1,32 +1,8 @@
+  // HALL LEVEL 8
   // ------------------
   {
     id: null,
```
Ids need to be set

cristian-aldea replied:
Thank Jon for setting them!

bit172 commented on src/components/map-overlays/map-overlays.component.tsx:
```diff
@@ -0,0 +1,137 @@
+  }, []);
+
+  return (
+    <>
```
Need comments to know what overlays these are

cristian-aldea replied:
Done! Good point

osahtout committed:
Fix ios bug for markers and polygons #19

While testing, we were running into issues with ios where the markers
and polygons weren't being rendered correctly. This issue has been
addressed by creating a new component for polygons and replacing the use
of index for keys

bit172 committed:
Replace Indoor id with uuid and fix types #19

This commit fixes type definitions for indoor location and added type information
for the custom polygon components.

bit172 reviewed (approved):
LGTM!

leojrsilao reviewed (approved):
Looks good on iOS now!

r-saba reviewed (approved):
Looks good on Pixel 2

RGPosadas reviewed (approved):
Documentation looks good!
